torch/csrc/jit/passes/bailout_graph.cpp

```diff
@@ -0,0 +1,203 @@
+      auto new_const = copy_graph_->createClone(node, {nullptr});
+      copy_graph_->block()->appendNode(new_const);
+      return new_const->output();
+    }
+
+    if (!new_value) {
+      new_value = copy_graph_->block()->addInput();
+      live_inputs_.push_back(old_value);
+    }
+    this->old_to_new_[old_value] = new_value;
+    new_value->copyMetadata(old_value);
+    return new_value;
+  }
+
+  void buildBailOutBlockFrom(Node* n) {
+    auto outer_node = n->owningBlock()->owningNode();
+    auto* block = copy_graph_->block();
+    if (n->kind() == prim::Loop) {
+      auto new_max_count = addNewInputForValue(n->inputs()[0]);
+      auto cur_iter = addNewInputForValue(n->blocks()[0]->inputs()[0]);
+      // subtract the number of iterations we already did
+      auto updated_max_trip_count = copy_graph_->create(aten::sub);
+      block->appendNode(updated_max_trip_count);
+      updated_max_trip_count->addInput(new_max_count);
+      updated_max_trip_count->addInput(cur_iter);
+      addNewInputForValue(n->inputs()[0], updated_max_trip_count->output());
+      // N.B. the rest of inputs have already been mapped
+      // when loop->blocks()[0] was processed
+    } else if (n->kind() == prim::If) {
+      // N.B. nothing to do; outputs should've already been mapped properly
+      // when we processed if block (either true of false arm)
+      // skip if since the rest of true or false block has already been
+      // inlined
+      n = n->next();
+    }
+
+    auto b = n->owningBlock();
+    // start from a given node, it could be any node anywhere in a graph
+    // if it happens to be an if node, we advance past it (see the above)
+    graph_node_list_iterator it(n, kNextDirection);
+    for (; it != b->nodes().end(); it++) {
+      auto env = [this](Value* v) {
+        auto new_value =
+            (this->old_to_new_.count(v) == 0) ? nullptr : this->old_to_new_[v];
+        return addNewInputForValue(v, new_value);
+      };
+      auto node = *it;
+
+      auto new_node = block->appendNode(copy_graph_->createClone(node, env));
+      for (size_t i = 0; i < node->outputs().size(); ++i) {
+        auto oo = node->outputs()[i];
+        auto no = new_node->outputs()[i];
+        old_to_new_[oo] = no;
+        no->copyMetadata(oo);
+      }
+    }
+
+    if (outer_node) {
+      auto block_outputs = n->owningBlock()->outputs();
+      // skip the first input for loops (current iteration count)
+      size_t i = outer_node->kind() == prim::Loop;
+      auto new_outputs = outer_node->kind() == prim::Loop
+          ? outer_node->inputs()
+          : outer_node->outputs();
+      for (; i < block_outputs.size(); i++) {
+        auto nv = old_to_new_[block_outputs[i]];
+        old_to_new_[new_outputs.at(i)] = nv;
+      }
+      buildBailOutBlockFrom(outer_node);
+    }
+  }
+
+  std::shared_ptr<Graph> buildBailOutGraphFrom(Node* n) {
+    old_to_new_.clear();
+    copy_graph_ = std::make_shared<Graph>();
```

Comment:
I feel like initialization should be done in the constructor.

Reply:
oh yeah... missed this... it's already designed that way. will move it to the c-tor